### ISSUE TITLE
Production use (or rather validation for adopting in production architecture)

### ISSUE THREAD

**Issue by drmikecrowe** [tidwall/tile38#288]
I'm liking everything I see about Tile38, and it looks like it will solve a technical issue I'd have to implement (roaming geofences).  Is there a list available of existing sites using Tile38?

I'm mainly seeking data to validate this choice:

* Who is using it in production?  
* What level of traffic are they handling?
* What don't I know yet that I need to know before depending on Tile38 in production?  (assuming I have no issues implementing Replication, failover, load-balance the reads, etc)

**Comment by dinapappor** [comment]
At work we're using tile38 with a master and 4 followers.

We're having about ~20 SETs at peak and about 1800 nearby at peak. We're running these on t2.mediums on aws and we're barely seeing any cpu usage at all. It's been 99-100% cpu idleness looking at cpu graphs for those instances. This goes for both master/leader and the followers.

We're doing loadbalancing by putting the followers behind an ELB that our API call. We're using redis interface and the amount of connections are fairly well spread over the followers. 

Doing failover is not something I have done yet. I have found no viable option for this yet. :(

**Comment by tobilg** [comment]
@dinapappor Thanks for the insights! Do you happen to also use the `hooks` functionalities, and if so, in what setting? Tile38 directly doing http(s) requests, or some layer in between?

**Comment by tobilg** [comment]
@tidwall I'd be great to eventually also have the production recommendations either in the readme or the tile38.com website. Thanks!

**Comment by tidwall** [comment]
@tobilg I agree. I'd be happy to explore ideas around how a recommendations page might best be organized. I'm often in the weeds with the codebase and I'm sure I take for granted the challenges with deploying in production.

Perhaps be new topic page on the tile38.com site?

**Comment by dinapappor** [comment]
@tobilg No, I do not use any hooks, communication is purely done by means of the redis interface. 

Changing back somewhat to the original topic:
I am curious as how to achieve redundancy and failovering with tile38. If anyone could give insights on that I would very much appreciate it.

**Comment by gmonk63** [comment]
@tobilg  Sorry if this is off topic but curious to know if your using geofencing and how you would go about getting notifications without web hooks.  I'm currently testing with python and Raspberry pi  with local mqtt broker.  Im setting gps coords with python-redis  and sending the entry,exit notifications to the local broker via mqtt hook. My code subscribes to the local broker to get those notifications.  but i cant see this as be being the best way..

Thanks

**Comment by tobilg** [comment]
@tidwall I think it would be great to have a dedicated page on [tile38.com](tile38.com) covering production deployments and operations, as well as some "recipes" on how to potentially scale out etc.

**Comment by KrishnaPG** [comment]
Came here while looking for tips on how to organize the collections for optimal performance in production. For example, if an object belongs to multiple categories (e.g. vehicle, and coldstorage categories), should these two categories be made two collections and enter the object into those two collections separately, or just set one object and maintain the categories as array field? Which one is more optimal for queries (e.g. select all vehicles in this radius etc.)

**Comment by trendstate** [comment]
Would be keen to see how this production deployments work. 😊 
I'm working through some sensitivity testing myself to work out how to make it as fast as possible -but I'm sure I'm barely scratching the surface
Thanks
